src/com/intellij/advancedExpressionFolding/PropertyUtil.java

```diff
@@ -0,0 +1,29 @@
+package com.intellij.advancedExpressionFolding;
+
+import org.jetbrains.annotations.NotNull;
+
+public class PropertyUtil {
+
+    @NotNull
+    public static String guessPropertyName(@NotNull String text) {
+        StringBuilder sb = new StringBuilder();
+        if (text.startsWith("get")) {
+            sb.append(text.substring(3));
```

Comment:
https://docs.oracle.com/javase/8/docs/api/java/lang/StringBuilder.html#append-java.lang.CharSequence-int-int- may be better here and below, it avoids an unnecessary string object creation.

Reply:
I just extracted the method but as you proposed I added the optimization in e9ceb2a 
If you'll have a moment please review. Thank you

Reply:
looks good, thanks!